core/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineRequest.java

```diff
@@ -135,15 +136,34 @@ static Parsed parseWithPipelineId(String pipelineId, Map<String, Object> config,
         if (pipeline == null) {
             throw new IllegalArgumentException("pipeline [" + pipelineId + "] does not exist");
         }
-        List<IngestDocument> ingestDocumentList = parseDocs(config);
-        return new Parsed(pipeline, ingestDocumentList, verbose);
+        boolean success = false;
+        try {
+            List<IngestDocument> ingestDocumentList = parseDocs(config);
+            Parsed parsed = new Parsed(pipeline, ingestDocumentList, verbose);
+            success = true;
+            return parsed;
+        } finally {
+            if (success == false) {
+                IOUtils.closeWhileHandlingException(pipeline);
```

Comment:
Why is this closing the pipeline and swallowing any failure? First, closing the pipeline here means any future requests to use the pipeline will fail? Is there any way for a user to get out of that situation? But if we do need to close, I think we should propagate any close problems?